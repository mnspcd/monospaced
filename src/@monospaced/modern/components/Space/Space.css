.mdrn .Space {
  margin-bottom: calc(1em * var(--leading));
}

.mdrn .Space--l {
  margin-bottom: calc(1em * var(--leading-scale-xl));
}

@media (--larger-viewport) {
  .mdrn .Space--l {
    margin-bottom: calc(1em * var(--leading-scale-xxl));
  }
}

.mdrn .Space--s {
  margin-bottom: calc(1em * var(--leading) / 2);
}

.mdrn .Space--xl {
  margin-bottom: calc(1em * var(--leading-scale-xxl));
}

@media (--larger-viewport) {
  .mdrn .Space--xl {
    margin-bottom: calc(1em * var(--leading-scale-xxxl));
  }
}
