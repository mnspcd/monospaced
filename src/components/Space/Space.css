.Space {
  margin-bottom: calc(1em * var(--leading));
}

.Space--l {
  margin-bottom: calc(1em * var(--leading-scale-xl));
}

@media (--larger-viewport) {
  .Space--l {
    margin-bottom: calc(1em * var(--leading-scale-xxl));
  }
}

.Space--s {
  margin-bottom: calc(1em * var(--leading) / 2);
}

.Space--xl {
  margin-bottom: calc(1em * var(--leading-scale-xxl));
}

@media (--larger-viewport) {
  .Space--xl {
    margin-bottom: calc(1em * var(--leading-scale-xxxl));
  }
}

.Space:last-child {
  margin-bottom: 0;
}
