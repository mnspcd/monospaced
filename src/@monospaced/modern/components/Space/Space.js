import classNames from "classnames";
import PropTypes from "prop-types";
import React from "react";

import "./Space.css";

/**
 * Use `Space` to create vertical space between components.
 */
const Space = ({ size = "m" }) => (
  <div
    className={classNames({
      Space: true,
      [`Space--l`]: size === "l",
      [`Space--s`]: size === "s",
      [`Space--xl`]: size === "xl",
    })}
  />
);

const sizes = ["s", "m", "l", "xl"];

Space.sizes = sizes;

Space.propTypes = {
  /**
   *  Size of the Space.
   */
  size: PropTypes.oneOf(sizes),
};

export default Space;
