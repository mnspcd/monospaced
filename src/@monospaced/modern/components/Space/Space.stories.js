import { select } from "@storybook/addon-knobs";
import React from "react";

import Block from "../Block";

import Space from "./Space";

export default {
  component: Space,
  title: "Components/Space",
};

export const basic = () => (
  <>
    <Block>Component</Block>
    <Space />
    <Block>Component</Block>
  </>
);

export const small = () => (
  <>
    <Block>Component</Block>
    <Space size="s" />
    <Block>Component</Block>
  </>
);

export const large = () => (
  <>
    <Block>Component</Block>
    <Space size="l" />
    <Block>Component</Block>
  </>
);

export const extraLarge = () => (
  <>
    <Block>Component</Block>
    <Space size="xl" />
    <Block>Component</Block>
  </>
);

export const knobs = () => (
  <>
    <Block>Component</Block>
    <Space size={select("size", [""].concat(Space.sizes)) || null} />
    <Block>Component</Block>
  </>
);

knobs.story = { parameters: { docs: { disable: true } } };
