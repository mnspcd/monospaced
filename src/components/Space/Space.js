import PropTypes from "prop-types";
import React from "react";
import classNames from "classnames";

import "./Space.css";

const Space = ({ children, size }) => (
  <div
    className={classNames({
      Space: true,
      [`Space--l`]: size === "l",
      [`Space--s`]: size === "s",
      [`Space--xl`]: size === "xl",
    })}
  >
    {children}
  </div>
);

Space.sizes = ["l", "s", "xl"];

Space.propTypes = {
  children: PropTypes.node.isRequired,
  size: PropTypes.oneOf(Space.sizes),
};

export default Space;
