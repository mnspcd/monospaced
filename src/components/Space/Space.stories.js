import React from "react";
import { select } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";

import Space from "./Space";
import Block from "../Block";

storiesOf("Components/Space", module).add("Component", () => {
  return (
    <React.Fragment>
      <Space size={select("size", [""].concat(Space.sizes)) || null}>
        <Block>Component</Block>
      </Space>
      <Block>Component</Block>
    </React.Fragment>
  );
});
