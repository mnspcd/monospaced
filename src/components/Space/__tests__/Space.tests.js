/* eslint-env jest */
import React from "react";
import { render } from "enzyme";

import Space from "../Space";

describe("Space component", () => {
  it("should render correctly", () => {
    const component = render(<Space>children</Space>);
    expect(component).toMatchSnapshot();
  });
});
