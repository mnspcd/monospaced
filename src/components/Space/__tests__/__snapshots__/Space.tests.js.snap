// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Space component should render correctly 1`] = `
<div
  class="Space"
>
  children
</div>
`;
